### PR TITLE
Depenedency Tree Reduction

### PR DESCRIPTION
**Pull request by cwfitzgerald**
Three main things happen in this PR: 

- I moved from having bve-build depending on cbindgen, to manually calling the executable.
- Moved from structopt to pico-args. This reduced dependencies signifigantly.
- Trimmed down dep tree in various ways.

Currently building the C API is disabled on mac due to install issues.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,7 +23,7 @@ debug = true
 inherits = "release"
 debug = true
 debug-assertions = false
-lto = true
+lto = "thin"
```
Comment (DethRaid) [inline Cargo.toml:26]: rhi thicc lto

Comment (cwfitzgerald) [reply Cargo.toml:26]: From what I can tell thin and fat lto perform about the same, but thin lto does lto in parallel. But just to be safe, release builds do fat lto.

File: bve-build/src/main.rs [hunk 2]
```diff
@@ -142,70 +121,41 @@ fn build(options: &Options) {
...
-        match result {
-            Ok(bindings) => {
-                bindings.write_to_file("bve-native/include/bve.hpp");
-            }
-            Err(err) => handle_cbindgen_error(&err, options),
-        }
-    }
+fn generate_c_bindings() {
+    println!("Generating C Bindings... (may take a while)");
+    let mut c = Command::new("cbindgen")
```
Comment (DethRaid) [inline bve-build/src/main.rs:126]: You're using command-line cbindgen instead of compiling it yourself? Sounds dope

Comment (cwfitzgerald) [reply bve-build/src/main.rs:126]: Yup, this is how I get <2s build times

File: bve-corpus/Cargo.toml [hunk 3]
```diff
@@ -15,18 +15,16 @@ keywords = ["bve", "serialization", "simulation"]
...
 num_cpus = "1.12.0"
+pico-args = "0.3.1"
 serde = { version = "1.0.104", features = ["derive"] }
 serde_json = "1.0.45"
-structopt = "0.3.8"
 tracing = "0.1.12"
-walkdir = "2.3.1"
-winapi = { version =  "0.3.8", features = ["processthreadsapi"] }
\ No newline at end of file
+walkdir = "2.3.1"
```
Comment (DethRaid) [inline bve-corpus/Cargo.toml:30]: add trailing newline

**Pull request merged**